pkg/kv/kvserver/allocator/mma/load.go

```diff
@@ -309,10 +313,35 @@ func (mm *meansMemo) getMeans(expr constraintsDisj) *meansForStoreSet {
 	}
 	means.constraintsDisj = expr
 	mm.constraintMatcher.constrainStoresForExpr(expr, &means.stores)
+	computeMeansForStoreSet(means.stores, mm.loadInfoProvider, means, mm.scratchNodes)
+	return means
+}
+
+// getStoreLoadSummary returns the load summary for a store in the context of
+// the given set (encoded in means). It attempts to utilize a cached value if
+// curLoadSeqNum permits.
+func (mm *meansMemo) getStoreLoadSummary(
+	means *meansForStoreSet, storeID roachpb.StoreID, curLoadSeqNum uint64,
+) storeLoadSummary {
+	summary, ok := means.tryGetStoreLoadSummary(storeID, curLoadSeqNum)
+	if ok {
+		return summary
+	}
+	summary = mm.loadInfoProvider.computeLoadSummary(storeID, &means.storeLoad, &means.nodeLoad)
+	means.putStoreLoadSummary(storeID, summary)
+	return summary
+}
+
+func computeMeansForStoreSet(
```

Comment:
Add code TODO for excluding stores based on node failure detection from the mean.